### DISCUSSION TITLE
Why not always set global.prisma in production?

### DISCUSSION THREAD

**Discussion by aheidelberg** [prisma/prisma#13240] [Q&A] [answered]
https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#prevent-hot-reloading-from-creating-new-instances-of-prismaclient

> As a workaround, you can store PrismaClient as a global variable in development environments only, as global variables are not reloaded

I understand this workaround to prevent multiple instances when hot reloading. But why is this recommended only for the dev environment? Why not just always set `global.prisma`, export that, and be done with it, save the extra `if` statement. Is there any disadvantage to setting `global.prisma` in a production environment?

**Comment by austincrim** [comment, marked as answer]
Hey @aheidelberg 👋 ,

Using global variables in production is generally discouraged to prevent variable name collisions, but you are free to do so, if you are willing to accept some risk.

If you do run into issues, feel free to come back and update this discussion!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
